s3torchconnector/src/s3torchconnector/_s3dataset_common.py

```diff
@@ -42,22 +38,18 @@ def parse_s3_uri(uri: str) -> Tuple[str, str]:
 
 def get_objects_from_uris(
     object_uris: Union[str, Iterable[str]], client: S3Client
-) -> Iterable[S3Reader]:
+) -> Iterable[S3BucketKey]:
     if isinstance(object_uris, str):
         object_uris = [object_uris]
     # TODO: We should be consistent with URIs parsing. Revise if we want to do this upfront or lazily.
-    bucket_key_pairs = [parse_s3_uri(uri) for uri in object_uris]
+    bucket_key_pairs = [S3BucketKey(*parse_s3_uri(uri)) for uri in object_uris]
```

Comment:
This is unintentionally changing the behaviour of this method. This was supposed to return an iterable but currently returning a list. The issue is not with this PR but actual implementation in the first place. Lets have another look at this part.